types/query/collections_pagination.go

```diff
@@ -237,13 +250,18 @@ func collFilteredPaginateByKey[K, V any, C Collection[K, V]](
 		// if no predicate is specified then we just append the result
 		if predicateFunc == nil {
 			results = append(results, kv)
-			count++
 			// if predicate is applied we execute the predicate function
 			// and append only if predicateFunc yields true.
-		} else if predicateFunc(kv.Key, kv.Value) {
-			results = append(results, kv)
-			count++
+		} else {
+			include, err := predicateFunc(kv.Key, kv.Value)
+			if err != nil {
+				return nil, nil, err
+			}
+			if include {
+				results = append(results, kv)
+			}
 		}
+		count++
```

Comment:
I had to make some adjustments to collections paginate function to better match the pagination behaviour